lib/internal/http2/core.js

```diff
@@ -1833,21 +1852,25 @@ class Http2Stream extends Duplex {
     session[kMaybeDestroy]();
     process.nextTick(emit, this, 'close', code);
     callback(err);
-  }
 
+  }
   // The Http2Stream can be destroyed if it has closed and if the readable
   // side has received the final chunk.
   [kMaybeDestroy](error, code = NGHTTP2_NO_ERROR) {
-    if (error == null) {
-      if (code === NGHTTP2_NO_ERROR &&
-          (!this._readableState.ended ||
-          !this._writableState.ended ||
-          this._writableState.pendingcb > 0 ||
-          !this.closed)) {
-        return;
-      }
+    if (error || code !== NGHTTP2_NO_ERROR) {
+      this.destroy(error);
+      return;
+    }
+
+    // TODO mcollina: remove usage of _*State properties
```

Comment:
This `TODO` is still in the previous style ;)

Reply:
done.